lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -705,8 +724,13 @@ func (d *zfs) CreateVolumeFromCopy(vol Volume, srcVol Volume, copySnapshots bool
 		revert.Add(func() { _ = d.DeleteVolume(vol, op) })
 	}
 
+	zfsVolumeType, err := d.getDatasetProperty(d.dataset(vol, false), "type")
```

Comment:
We should be able to use the srcVol DB config to decide this and avoid running the extra command.